Sources/StreamChat/Controllers/ChannelListController/ChannelListController.swift

```diff
@@ -196,6 +220,33 @@ public class ChatChannelListController: DataController, DelegateCallable, DataSt
     public func setDelegate<Delegate: ChatChannelListControllerDelegate>(_ delegate: Delegate) {
         multicastDelegate.mainDelegate = AnyChannelListControllerDelegate(delegate)
     }
+    
+    private func handleUpdatedChannels(_ changes: [ListChange<ChatChannel>]) {
+        let channels = changes.compactMap { change -> ChatChannel? in
+            switch change {
+            case let .insert(channel, _):
+                return (delegate?.controller(self, shouldLinkNewChannel: channel) ?? false) ? channel : nil
+            case let .update(channel, _):
+                return (delegate?.controller(self, shouldLinkUpdatedChannel: channel) ?? false) ? channel : nil
+            default: return nil
+            }
+        }
```

Comment:
Should we wrap this with `delegateCallback` to make sure the delegate is invoked on callback queue?

Reply:
IMHO it's not needed, the same reason `completion` is not needed for this func: users should keep the processing to a minimum, we don't want them to make an API call for this, it'd just make stuff more complex